doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,1060 @@
+
+By providing VLAN-to-instance mapping, MSTP ensures that a limited number of instances are created, allowing for network segregation. In contrast, PVST lacks control over VLAN-to-instance mapping, leading to numerous instances and inefficient memory utilization. 
+
+STP only supports a single instance, resulting in under utilized network bandwidth due to complete link blocking. Therefore, MSTP offers better control over instances, efficient resource usage, and improved network performance compared to PVST and STP.
+
+MSTP calculates spanning trees on the basis of Multiple Spanning Tree Bridge Protocol Data Units (MST BPDUs).
+
+<div align="center">
+<img src="images/MSTP_BPDU.png" alt="MSTP BPDU">
+<p>MSTP BPDU Format</p>
+</div><br>
+<div align="center">
+<img src="images/MSTI_config_message.PNG" alt="MSTP Config Message">
+<p>MSTI Configuration Messages</p>
+</div>
+
+*Refer to [RFC IEEE 802.1s-2002](https://standards.ieee.org/ieee/802.1s/1042/) for MSTP BPDU details.* 
+
+# Requirements
+1. Support the creation of Multiple Spanning Tree Instances (MSTIs).
+1. Support the assignment of one or more VLANs to a specific MSTI within a region.
+1. Support the option to assign a region name and revision number to MSTP regions in order to achieve unique identification of VLAN to instance mapping across switches.
+1. Support path selection and forwarding behaviour in MSTI to optimize network performance within each instance by configuring a distinct root bridge.
+1. Support the configuration of spanning tree parameters such as forward delay, hello timer, hop count and max age.
+1. The Destination Mac Address will be 01:80:C2:00:00:00 for MSTP BPDUs.
+1. Support compatibility with networks employing different spanning tree protocols, such as STP, RSTP and PVST via Protocol Migration. 
+1. Support edge port functionality
+1. Support BPDU guard functionality
+1. Support root guard functionality
+1. Support protocol operation on static breakout ports
+1. Support protocol operation on Port-channel interfaces
+
+# Architecture Design
+Following diagram explains the architectural design and linkages for MSTP. MSTP uses multiple existing SONiC containers, configuration details of each is mentioned below as well.
+
+![MSTP Architecture](images/MSTP_architecture.png)
+
+## STP Container
+STP Container is responsible for actions taken for BPDU rx and BPDU tx. Following are the details for implementation:
+
+### STPMgr
+Subscribes to CONFIG_DB and STATE_DB tables, parsing configurations and passes to STPd.
+
+### STPd
+Responsible for all MST protocol related calculations. BPDUs are sent and received in STPd and states are updated accordingly.
+
+### STPSync
+A process running as a part of STPD. Responsible for updating all the MSTP states in APP DB.
```

Comment:
Couple of questions here, 

1) is "stpSync" process forked within "STPd". 
2) Is "STPd" taken from a open-source implementation for instance https://github.com/sonic-net/sonic-stp ?

Assuming the answer to the above is 'yes', would be it cleaner to have "STPsync" implemented outside of "STPd" and then have IPCs to communicate port-states, fdb flush & link state events. 

Doing this will ensure "STPd" repo has minimal integration changes and syncing it later to open-source repo would be easier.

Reply:
stpsync is not a process. STPSync is a component integrated within STPd responsible for managing and updating all operational STP data to APP DB. 

We will update the same in HLD.